### PR TITLE
User redirection

### PR DESCRIPTION
**Pull request by KamilDonda**
Redirecting the user from pages that are not accessible:
+ **Scientific Works** -> Sign In as Participant
+ **My Reviews** -> Sign In as Reviewer
+ **Adding Work** -> Home (if we are participant and we added work) or Sign In as Participant (otherwise)
+ **Work View** -> Home (if we are logged) or Sign In as Participant (unlogged)

### CODE REVIEW

File: web-client/src/pages/ScientificWorks/scientific-works.js [hunk 1]
```diff
@@ -60,21 +71,25 @@ export default function ScientificWorks() {
...
     if (!date) return null;
-    return date.replace(
-      /(\d{4})-(\d{1,2})-(\d{1,2})/,
-      function (_, y, m, d) {
-        return d + "/" + m + "/" + y;
-      }
-    );
+    return date.replace(/(\d{4})-(\d{1,2})-(\d{1,2})/, function (_, y, m, d) {
+      return d + "/" + m + "/" + y;
+    });
```
Comment (JustD4nTe) [inline web-client/src/pages/ScientificWorks/scientific-works.js]: We don't needed this function anymore

File: web-client/src/pages/ScientificWorks/scientific-works.js [hunk 2]
```diff
@@ -88,63 +103,71 @@ export default function ScientificWorks() {
...
-        {/* If list of works is null, then nothing is displayed */}
-        {works[0]?.title ? workList : null}
-      </div>
+    isSuccessedLoad && (
+      <div className={style.main}>
+        <h1>Scientific works</h1>
+        <div className={style.works}>
+          <div className={style.list}>
+            {/* If list of works is null, then nothing is displayed */}
+            {works[0]?.title ? workList : null}
```
Comment (JustD4nTe) [inline web-client/src/pages/ScientificWorks/scientific-works.js]:
```suggestion
            {works[0]?.title && workList}
```

File: web-client/src/pages/Users/Reviewer/my-reviews.js [hunk 3]
```diff
@@ -1,83 +1,103 @@
...
+
+    (async () => {
+      await axios
+        .get(`${URL_API}/Reviewer/MyReviews`, {
+          headers: { Authorization: `Bearer ${token}` },
+        })
+        .then((resp) => {
+          SetIsSuccessedLoad(true);
+          SetReviews(resp.data);
+          console.log(resp.data);
```
Comment (JustD4nTe) [inline web-client/src/pages/Users/Reviewer/my-reviews.js]: Remove console.log

File: web-client/src/pages/Users/Reviewer/my-reviews.js [hunk 4]
```diff
@@ -1,83 +1,103 @@
...
+  }, [location, history]);
+
+  // convert date from ISO [YYYY-MM-DDTHH:mm:ss.sssZ] to DD/MM/YYYY
+  const convertDate = (date) => {
+    date = date?.substring(0, 10);
+    if (!date) return null;
+    return date.replace(/(\d{4})-(\d{1,2})-(\d{1,2})/, function (_, y, m, d) {
+      return d + "/" + m + "/" + y;
+    });
+  };
```
Comment (JustD4nTe) [inline web-client/src/pages/Users/Reviewer/my-reviews.js]: We don't need this function too

**Review by JustD4nTe — APPROVED** [review]

**Review by Kwolik — CHANGES_REQUESTED** [review]
Look address page.  Generally unsecured site my-profie = reviewer = participant
image
image
image


Page for not logged in. Change
image

**Comment by JustD4nTe** [comment]
@Kwolik My profile is not implemented yet, so there is no way to check if user can see this page
> Look address page. Generally unsecured site my-profie = reviewer = participant

@Kwolik What do you want to change?
> Page for not logged in. Change

**Review by upsu90 — CHANGES_REQUESTED** [review]
image
If I'm not logged in and go to _my reviews, adding work_, I get error 404 page

If I'm logged as Participant and go to _my reviews, adding work_, I get error 404 page

If I'm logged as Reviewer and go to _adding work_, I get error 404 page

**Review by Kwolik — COMMENTED** [review]
@JustD4nTe The page with link approval should be only for people who are not logged in

**Review by upsu90 — APPROVED** [review]

**Review by Kwolik — APPROVED** [review]

**Pull request merged**